vllm/v1/engine/output_processor.py

```diff
@@ -438,6 +458,11 @@ def add_request(
         if parent_req:
             self.parent_requests[parent_req.request_id] = parent_req
 
+        # Track the external_req_id -> [internal_req_id, ...] mapping
+        if req_state.external_req_id not in self.external_req_ids:
+            self.external_req_ids[req_state.external_req_id] = []
```

Comment:
Could consider using a `defaultdict(list)` for this, then these lines wouldn't be needed.